base/compiler/optimize.jl

```diff
@@ -509,7 +711,7 @@ function run_passes(
     @pass "compact 2" ir = compact!(ir)
```

Comment:
This pass currently seems to not be IPO safe, since the access to `ci.propagate_inbounds` (and `:boundscheck` removal folding) is not currently IPO safe

Reply:
Yes, I'm working on that right now.